payu/models/access.py

```diff
@@ -52,13 +52,24 @@ def __init__(self, expt, name, config):
             if model.model_type == 'cice5':
                 model.access_restarts.append(['u_star.nc', 'sicemass.nc'])
 
+            if model.model_type == 'cice':
+                # Structure of model coupling namelist
+                model.cpl_fname = 'input_ice.nml'
+                model.cpl_group = 'coupling'
+                model.runtime0_key = 'runtime0'
+                model.start_date_nml_name = "restart_date.nml"
+
+            if model.model_type == 'matm':
+                # Structure of model coupling namelist
+                model.cpl_fname = 'input_atm.nml'
+                model.cpl_group = 'coupling'
+                model.runtime0_key = 'truntime0'
+                model.start_date_nml_name = "restart_date.nml"
+
     def setup(self):
         if not self.top_level_model:
             return
 
-        cpl_keys = {'cice': ('input_ice.nml', 'coupling', 'runtime0'),
-                    'matm': ('input_atm.nml', 'coupling', 'truntime0')}
-
```

Comment:
I moved these from here to the initialisation section, making them model attributes so that they would be accessible when writing the new restart date during `archive`.